packages/contracts-bedrock/scripts/checks/semver-natspec/main.go

```diff
@@ -0,0 +1,215 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"strings"
+	"sync"
+	"sync/atomic"
+)
+
+type ArtifactsWrapper struct {
+	RawMetadata string `json:"rawMetadata"`
+}
+
+type Artifacts struct {
+	Output struct {
+		Devdoc struct {
+			StateVariables struct {
+				Version struct {
+					Semver string `json:"custom:semver"`
+				} `json:"version"`
+			} `json:"stateVariables,omitempty"`
+			Methods struct {
+				Version struct {
+					Semver string `json:"custom:semver"`
+				} `json:"version()"`
+			} `json:"methods,omitempty"`
+		} `json:"devdoc"`
+	} `json:"output"`
+}
+
+var ConstantVersionPattern = regexp.MustCompile(`string.*constant.*version\s+=\s+"([^"]+)";`)
+
+var FunctionVersionPattern = regexp.MustCompile(`^\s+return\s+"((?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)(?:-(?P<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?P<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)";$`)
+
+var InteropVersionPattern = regexp.MustCompile(`^\s+return\s+string\.concat\(super\.version\(\), "((.*)\+interop(.*)?)"\);`)
+
+func main() {
+	if err := run(); err != nil {
+		writeStderr("an error occurred: %v", err)
+		os.Exit(1)
+	}
+}
+
+func writeStderr(msg string, args ...any) {
+	_, _ = fmt.Fprintf(os.Stderr, msg+"\n", args...)
+}
+
+func run() error {
+	cwd, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("failed to get current working directory: %w", err)
+	}
+
+	writeStderr("working directory: %s", cwd)
+
+	artifactsDir := filepath.Join(cwd, "forge-artifacts")
+	srcDir := cwd
+
+	artifactFiles, err := glob(artifactsDir, ".json")
+	if err != nil {
+		return fmt.Errorf("failed to get artifact files: %w", err)
+	}
+	contractFiles, err := glob(srcDir, ".sol")
```

Comment:
This could break if `glob` searches through `src` before `forge-artifacts`, since the `contractFiles` glob match will find a directory in `forge-artifacts` named `<ContractName>.sol` as well. `filepath.Ext` doesn't look to differentiate between files and directories, either. Right now it's working because `src` is read second, overwriting the first write to the `out` map in `glob(...)`.

```suggestion
	artifactsDir := filepath.Join(cwd, "forge-artifacts")
	srcDir := filepath.Join(cwd, "src")

	artifactFiles, err := glob(artifactsDir, ".json")
	if err != nil {
		return fmt.Errorf("failed to get artifact files: %w", err)
	}
	contractFiles, err := glob(srcDir, ".sol")
```

Reply:
this was an oversight, thank you!